gateway/src/inference/providers/anthropic.rs

```diff
@@ -1721,6 +1732,7 @@ mod tests {
             top_p: Some(0.9),
             tool_choice: None,
             tools: None,
+            disable_parallel_tool_use: None,
```

Comment:
can you add test cases in here where the ToolConfig is set with parallel_tool_calls=true and parallel_tool_calls=false?

Reply:
Good call, while the tests for true case were already there, we were missing on false behavior.
Added an explicit test to test both the behaviors.

Reply:
yep, will do one pass of reviews and then merge! great work